DylibBinder/DBFunc.cs

```diff
@@ -0,0 +1,123 @@
+		{
+			Name = tlf.Name.Name;
+			IsStatic = GetStaticStatus (tlf.Signature);
+			IsProperty = false;
+			HasThrows = tlf.Signature.CanThrow;
+			IsMutating = isMutating;
+			OperatorKind = tlf.Operator;
+			// TODO not sure if/how to handle instances with protocols yet
+			HasInstance = !tlf.Signature.IsConstructor && !IsStatic && !isProtocol;
+			ReturnType = SwiftTypeToString.MapSwiftTypeToString (tlf.Signature.ReturnType, tlf.Module.Name);
+			ParameterLists = new DBParameterLists (tlf.Signature, HasInstance);
+			GenericParameters = new DBGenericParameters (tlf.Signature);
+		}
+
+		public DBFunc (DBProperty dbProperty, string propertyType, bool isMutating = false)
+		{
+			if (propertyType == "Getter")
+				Name = $"get_{dbProperty.Name}";
+			else
+				Name = $"set_{dbProperty.Name}";
+			PropertyType = propertyType;
+			IsStatic = dbProperty.IsStatic;
+			IsProperty = true;
+			HasThrows = false;
+			IsMutating = isMutating;
+			HasInstance = !IsStatic;
+			ReturnType = dbProperty.Type;
+			ParameterLists = new DBParameterLists (ReturnType, HasInstance, propertyType);
+			GenericParameters = dbProperty.GenericParameters;
+		}
+
+		public string Name { get; }
+		public bool IsStatic { get; }
+		public bool IsProperty { get; }
+		public string ReturnType { get; }
+		public bool HasThrows { get; }
+		public bool IsMutating { get; }
+		public bool HasInstance { get; }
+
+		public string PropertyType { get; } = string.Empty;
+		public bool IsPossiblyIncomplete { get; } = false;
+		public OperatorType OperatorKind { get; } = OperatorType.None;
+		public TypeAccessibility Accessibility { get; } = TypeAccessibility.Public;
+		public bool IsFinal { get; } = false;
+		public bool IsDeprecated { get; } = false;
+		public bool IsUnavailable { get; } = false;
+		public bool IsOptional { get; } = false;
+		public bool IsRequired { get; } = false;
+		public bool IsConvenienceInit { get; } = false;
+		public string ObjcSelector { get; } = string.Empty;
+
+		public DBParameterLists ParameterLists { get; }
+		public DBGenericParameters GenericParameters { get; }
+		public List<DBAssociatedType> AssociatedTypes { get; } = new List<DBAssociatedType> ();
+
+		bool GetStaticStatus (SwiftBaseFunctionType signature)
+		{
+			switch (signature) {
+			case SwiftStaticFunctionType:
+				return true;
+			case SwiftPropertyType propType:
+				return propType.IsStatic;
+			case SwiftUncurriedFunctionThunkType:
+				return false;
+			case SwiftUncurriedFunctionType:
+				return false;
+			default:
+				return false;
+			}
+		}
+	}
+
+	internal class DBFuncs {
+		public DBFuncs (ClassContents classContents)
+		{
+			var functions = SortedSetExtensions.CreateOverloadSortedSet ();
+			functions.AddRange (classContents.Methods.Values, classContents.StaticFunctions.Values, classContents.Constructors.Values);
+
+			foreach (var function in functions) {
+				foreach (var overloadFunction in function.Functions) {
```

Comment:
This one seems kind of long to me.

If I did this right, I think it would turn into: 
```
Funcs.AddRange (functions.Where (t => t.Name.Name.IsPublic () && !IsMetaClass (t.Signature)).Select (t => new DBFunc (t, isProtocol: true)));
```

Opposed to the original: 
```
foreach (var function in functions) {
	if (function.Name.Name.IsPublic () && !IsMetaClass (function.Signature))
		Funcs.Add (new DBFunc (function, isProtocol: true));
}
```